experimental/dds/tree2/src/test/simple-tree/primitives.spec.ts

```diff
@@ -4,52 +4,131 @@
  */
 
 import { strict as assert } from "assert";
-import { leaf, SchemaBuilder } from "../../domains";
-import { getOldRoot, pretty } from "./utils";
-
-const _ = new SchemaBuilder({ scope: "test", libraries: [leaf.library] });
-const schema = _.intoSchema(_.optional(leaf.all));
-
-const testCases = [
-	undefined, // via optional root
-
-	// TODO: null,
-
-	true,
-	false,
-
-	-Infinity,
-	-Number.MAX_VALUE,
-	Number.MIN_SAFE_INTEGER,
-	-Number.MIN_VALUE,
-	-0,
-	NaN,
-	0,
-	Number.MIN_VALUE,
-	Number.MAX_SAFE_INTEGER,
-	Number.MAX_VALUE,
-	Infinity,
-
-	"", // empty string
-	"!~", // printable ascii range
-	"比特币", // non-ascii range
-	"😂💁🏼‍♂️💁🏼‍💁‍♂", // surrogate pairs with glyph modifiers
-];
+import { ImplicitFieldSchema } from "../../class-tree";
+// eslint-disable-next-line import/no-internal-modules
+import { InsertableTreeFieldFromImplicitField } from "../../class-tree/internal";
+import { getRoot, makeSchema, pretty } from "./utils";
 
 // Construct a SharedTree with each of the above primitives as the root and then
 // 'deepEquals' compares the proxy with the original primitive value.
 //
 // Also covers the corner case of an empty tree (via optional root) by constructing
 // a tree with an 'undefined' root.
 describe("Primitives", () => {
-	describe("satisfy 'deepEquals'", () => {
-		for (const initialTree of testCases) {
-			const proxy = getOldRoot(schema, initialTree);
-			const real = initialTree;
+	function checkExact<TSchema extends ImplicitFieldSchema>(
```

Comment:
Nit: it might be worth adding some brief docs to these helper functions for devs looking at them in the future.